src/main/kotlin/com/hibob/feedback/service/FeedbackService.kt

```diff
@@ -0,0 +1,31 @@
+package com.hibob.feedback.service
+
+import com.hibob.feedback.dao.*
+import org.springframework.http.HttpStatus
+import org.springframework.stereotype.Component
+import org.springframework.web.server.ResponseStatusException
+import java.util.*
+
+@Component
+class FeedbackService(private val feedbackDao: FeedbackDao, private val employeesDao: EmployeesDao) {
+
+    fun createFeedback(feedbackInput: FeedbackInput, activeUser: ActiveUser): UUID {
+        return feedbackDao.createFeedback(feedbackInput, activeUser)
+    }
+
+    fun getFeedback(feedbackId: UUID, activeUser: ActiveUser): Feedback? {
+        val feedback = feedbackDao.getFeedback(feedbackId, activeUser)
+        if (isNotAuthorized(employeesDao.getEmployeeByActiveUser(activeUser), feedback)) {
+            throw ResponseStatusException(
+                HttpStatus.UNAUTHORIZED,
+                "Unauthorized Access- Trying to fetch feedback which is not yours, while youre not HR or admin"
+            )
+        }
+        return feedback
```

Comment:
permission checks should be done in the resource layer

Reply:
also, you should be able to use the claims in the token to determine the permissions without needing to make a DB query

Reply:
so I need to add role and department into the activeUser data class that I receive in the claim?
and how can I check If the activeUser is the feedback writer without the DB?

Reply:
> so I need to add role and department into the activeUser data class that I receive in the claim?

yes, that would be the best

>  how can I check If the activeUser is the feedback writer without the DB?

you're right that you'll need to fetch the feedback, but you won't need to fetch the employee

Reply:
from the spec:
API Endpoint: Provide an API endpoint for HR/Admin to view all submitted feedback.

so I dont need anything else only role and department